src/public_id.rs

```diff
@@ -103,4 +107,78 @@ mod test {
         assert_eq!(relocated.client_name(), original_name);
         assert_eq!(relocated.name(), relocated_name);
     }
+
+    #[test]
+    fn test_calculate_relocated_name() {
```

Comment:
I'm sorry I gave you wrong information; this should be in ::utils (and it probably was there before ;) )  My bad;

If you can avoid/remove "test_" as a general improvement that's also great, thanks!